### PR TITLE
WIP: Add org.openchemistry.avogadro2

### PR DESCRIPTION
**Pull request by kevinsmia1939**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:** https://discuss.avogadro.cc/t/im-creating-flatpak-for-avogadro2-need-help-on-spglib/3456/17
- [ ] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by kevinsmia1939** [comment]
why flathubbot think it is a json file?

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by nedrichards** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by hfiguiere** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

**Review by hfiguiere — CHANGES_REQUESTED** [review]

File: org.openchemistry.Avogadro2.yaml [hunk 1]
```diff
@@ -0,0 +1,266 @@
...
+  - name: libxml2
+    sources:
+      - type: archive
+        url: http://xmlsoft.org/sources/libxml2-2.9.0.tar.gz
+        sha256: ad25d91958b7212abdc12b9611cfb4dc4e5cddb6d1e9891532f48aacee422b82  
+      - type: shell
+        commands:
+          - cp -p /usr/share/automake-*/config.{sub,guess} .;
+      
+  - name: libarchive
```
[the remark this answers is not in the document]
Comment (kevinsmia1939) [reply org.openchemistry.Avogadro2.yaml:54]: Thanks, I will remove it. How do I check if it already in runtime?

**Comment by kevinsmia1939** [comment]
bot, build org.openchemistry.Avogadro2

File: org.openchemistry.Avogadro2.yaml (hunk 1 above)
[the remark this answers is not in the document]
Comment (hfiguiere) [reply org.openchemistry.Avogadro2.yaml]: usually I try to build and add dependencies as needed. Or I look into the runtime itself. It's a lot of trial and effort.

**Comment by hfiguiere** [comment]
bot, build org.openchemistry.Avogadro2

**Comment by barthalion** [comment]
```
ScriptLoader::queryProgramName: Unable to retrieve program name for "/app/lib/avogadro2/scripts/formatScripts/cclibScript.py" ; "Error running script '/usr/bin/python3 /app/lib/avogadro2/scripts/formatScripts/cclibScript.py --display-name --lang en_DK': Abnormal exit status 1 (Unknown error.: Unknown error)\n\nOutput:\nTraceback (most recent call last):\n  File \"/app/lib/avogadro2/scripts/formatScripts/cclibScript.py\", line 22, in <module>\n    from cclib.io.ccio import ccopen\nModuleNotFoundError: No module named 'cclib'\n"
```

Not sure if it's a blocker for anything.

**Comment by barthalion** [comment]
Hey @ghutchis, would you or someone else working on Avogadro be interested in co-maintaning it on Flathub? There's no obligation to do so, but we have a rule about upstream involvement and we'd like you to have write access even when @kevinsmia1939 will be handling the maintenance.

**Comment by ghutchis** [comment]
> Not sure if it's a blocker for anything.

No, that's not a blocker. We need to clean up our warnings - we try to load Python scripts at that stage. (That particular script will be made an optional post-install one soon.)

**Comment by ghutchis** [comment]
> Hey @ghutchis, would you or someone else working on Avogadro be interested in co-maintaning it on Flathub? 

Sure, that's fine.

As a side question, is there a good way to make the Flatpak available as part of our GH actions builds? We've had some people frustrated with the official packages, and we'd like to add it to continuous release packaging.

**Comment by barthalion** [comment]
> As a side question, is there a good way to make the Flatpak available as part of our GH actions builds? We've had some people frustrated with the official packages, and we'd like to add it to continuous release packaging.

There are multiple ways to do that. You can either use https://github.com/flathub/flatpak-external-data-checker/ or "abuse" GitHub Actions as event bus like here:

https://github.com/flathub/io.openrct2.OpenRCT2/blob/master/.github/workflows/beta.yml
https://github.com/OpenRCT2/OpenRCT2/pull/12836/files

This is for pushing every commit to Flathub Beta which is generally discouraged but can be tuned to update Flathub when you push a new tag.

**Pull request closed without merging**